### PR TITLE
refactor: lint improvements + format

### PR DESCRIPTION
**Pull request by h1376h**
This PR tries to make things more readable by using lint suggestions, **without any breaking changes** and change in behavior.

Linter rules' list is selected from what [flutter](https://github.com/flutter/flutter/blob/master/analysis_options.yaml) and some other [packages](https://github.com/rrousselGit/riverpod/blob/master/all_lint_rules.yaml) and projects use and can be edited or modified however you want them to be.

Most of the changes are made by running `dart fix --apply`

_After rebasing, I tried to add almost all of the needed rules altogether, instead of minor PRs and I fixed some of the warnings that couldn't be fixed automatically._

Most of `// ignore: lint rule` parts aren't that much important, but there's [one](https://dart-lang.github.io/linter/lints/avoid_slow_async_io.html) in _downloadIsarCore method, located in `isar/lib/src/native/isar_core.dart` that needs a fix, but I tried not to break anything by fixing [these kind of lints](https://dart-lang.github.io/linter/lints/avoid_classes_with_only_static_members.html), so I just ignored them with comments for now.

### CODE REVIEW

**Comment by h1376h** [comment]
@leisim Can't see your comment.

**Review by leisim — CHANGES_REQUESTED** [review]

**Comment by leisim** [comment]
@h1376h sorry, my bad. didn't commit the review

File: packages/isar/analysis_options.yaml [hunk 1]
```diff
@@ -7,4 +7,169 @@ analyzer:
   language:
     strict-casts: true
     strict-inference: true
-    strict-raw-types: true
\ No newline at end of file
+    strict-raw-types: true
+    
+linter:
```
[the remark this answers is not in the document]
Comment (h1376h) [reply packages/isar/analysis_options.yaml:14]: > How did you choose which lints to use?

https://github.com/flutter/flutter/blob/master/analysis_options.yaml

I'm using the lints listed here, and the commented ones should be fixed later.
Riverpod also uses the default lints and I'm a litte bit inspired by that package and how it handles lints as well.


[very_good_analysis](https://pub.dev/packages/very_good_analysis) is a good alternative, but using the "default" ones listed in flutter repo isn't that much hard to maintain.

[the remark this answers is not in the document]
Comment (leisim) [reply packages/isar/analysis_options.yaml:14]: Makes total sense. Is this PR ready to merge then?

[the remark this answers is not in the document]
Comment (h1376h) [reply packages/isar/analysis_options.yaml:14]: > Makes total sense. Is this PR ready to merge then?

yeah, but please check before merging, most of the changes are made automatically by `dart fix --apply`, I reviewed most of the codes and I didn't see anything wrong.

**Comment by leisim** [comment]
@all-contributors please add @h1376h for code and maintenance!

**Review by leisim — APPROVED** [review]

**Comment by h1376h** [comment]
Your change request to uncomment that line is what need to happen, but I didn't fix the warnings for that one yet.

**Pull request merged**